distribution/lib/Standard/Test/0.0.0-dev/src/Main.enso

```diff
@@ -21,13 +25,58 @@ import Standard.Base.System
                      2*3 . should_equal 6
 Suite.run_main : Any -> Nothing
 Suite.run_main ~specs =
-    r = this.run specs here.config_from_env
+    config = here.config_from_env
+    r = here.wrap_junit_testsuites config <|
+        this.run specs config
     code = if r.is_fail then 1 else 0
     System.exit code
 
+
+## PRIVATE
+find_project_root : File -> File
+find_project_root path =
+    if path.is_nothing then Nothing else
+        handler _ = Nothing
+        Panic.catch NullPointerException handler=handler <|
+            if path.name == "src" then path.parent else
+                @Tail_Call here.find_project_root path.parent
+
+
+## PRIVATE
+find_caller_script : [Stack_Trace_Element] -> File
+find_caller_script stack =
+    find_main idx =
+        if stack.at idx . name == "Suite.run_main" then idx else
+            @Tail_Call find_main (idx + 1)
+    main_index = find_main 0
+
+    find_caller idx =
+        source = stack.at idx . source_location
+        if source.is_a Source_Location then stack.at idx . source_location . file else
+            @Tail_Call find_caller (idx + 1)
```

Comment:
This also feels like it should handle exceptional cases better.

Consider running from the REPL - then I think the `source_location` will be missing for the caller, so it will never stop and fail at some point when `idx` falls out of range of `stack`.

This may not be an issue though, because we cannot even import `Tests` in REPL currently.